bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -651,6 +661,43 @@ private Request sendRequestBuilder(String subUrl, HttpMethod method) {
                 .agent(TAHOMA_AGENT);
     }
 
+    /**
+     * Returns JSESSIONID cookie value
+     *
+     * @return
+     * @throws ExecutionException
+     * @throws TimeoutException
+     * @throws InterruptedException
+     */
+    private String loginCozytouch() throws InterruptedException, TimeoutException, ExecutionException {
+        String authBaseUrl = "https://" + COZYTOUCH_OAUTH2_URL;
+
+        String urlParameters = "grant_type=password&username=" + urlEncode(thingConfig.getEmail()) + "&password="
+                + urlEncode(thingConfig.getPassword());
+
+        ContentResponse response = httpClient.newRequest(authBaseUrl + COZYTOUCH_OAUTH2_TOKEN_URL)
+                .method(HttpMethod.POST).header(HttpHeader.ACCEPT_LANGUAGE, "en-US,en")
+                .header(HttpHeader.ACCEPT_ENCODING, "gzip, deflate").header("X-Requested-With", "XMLHttpRequest")
+                .header(HttpHeader.AUTHORIZATION, "Basic " + COZYTOUCH_OAUTH2_BASICAUTH)
+                .timeout(TAHOMA_TIMEOUT, TimeUnit.SECONDS).agent(TAHOMA_AGENT)
+                .content(new StringContentProvider(urlParameters), "application/x-www-form-urlencoded; charset=UTF-8")
+                .send();
+
+        SomfyTahomaOauth2Reponse oauth2response = gson.fromJson(response.getContentAsString(),
+                SomfyTahomaOauth2Reponse.class);
+
+        logger.debug("OAuth2 Access Token: {}", oauth2response.getAccess_token());
+
+        response = httpClient.newRequest(authBaseUrl + COZYTOUCH_OAUTH2_JWT_URL).method(HttpMethod.GET)
+                .header(HttpHeader.AUTHORIZATION, "Bearer " + oauth2response.getAccess_token())
+                .timeout(TAHOMA_TIMEOUT, TimeUnit.SECONDS).send();
+
+        String jwt = response.getContentAsString();
+
+        logger.debug("JWT value: {}", jwt);
```

Comment:
I would rather use `if (logger.isDebugEnabled())` since `getContentAsString()` may be expensive. Please consider it, I am not sure how long is the JWT response...

Reply:
the getContentAsString is mandatory to get the token that will be able to perform other authenticated calls. However, I change the logg to trace, with a conditional check on log level.